base/compiler/ssair/passes.jl

```diff
@@ -269,6 +269,15 @@ function is_pending(compact::IncrementalCompact, old::OldSSAValue)
     return old.id > length(compact.ir.stmts) + length(compact.ir.new_nodes)
 end
 
+function is_getfield_captures(@nospecialize(def), compact::IncrementalCompact)
+    isa(def, Expr) || return false
+    length(def.args) >= 3 || return false
+    is_known_call(def, getfield, compact) || return false
+    which = compact_exprtype(compact, def.args[3])
```

Comment:
Should we check the type of def.args[2] also, to skip the walk if it can't be an OC type?

Reply:
I guess that would make sense. I don't really have a good intuition for what would be faster.